### ISSUE TITLE
about the two methods

### ISSUE THREAD

**Issue by zhexinli** [hustzxd/LSQuantization#5]
Hello. It seems that there are 2 implementation method in your code, referred as method1 and method2.
https://github.com/hustzxd/LSQuantization/blob/86cf51193ec197ec607d271acaa488fd9a8ff60e/lsq.py#L80-L87
method1 is supported to be faster than method2 at broadcasting because instead of computing the gradients using self-defined ops with indication masks, it compute the grads with build-in method in pytorch. But I'm wondering that does method1 reach the same accuracy as method2 does? After all, the method1 looks very simple in the code.

**Comment by hustzxd** [comment]
The two methods are logically the same. Maybe the purpose of method 2 is to easily understand the calculation process of backpropagation.

**Comment by zhexinli** [comment]
Got it! Thank you!
